services/port_manager/src/main/java/com/futurewei/alcor/portmanager/processor/PortProcessor.java

```diff
@@ -0,0 +1,239 @@
+        List<PortEntity.ExtraDhcpOpt> oldExtraDhcpOpts = oldPortEntity.getExtraDhcpOpts();
+
+        if (newExtraDhcpOpts != null && !newExtraDhcpOpts.equals(oldExtraDhcpOpts)) {
+            oldPortEntity.setExtraDhcpOpts(newExtraDhcpOpts);
+        }
+    }
+
+    private void updateAllowedAddressPairs(PortEntity newPortEntity, PortEntity oldPortEntity) {
+        List<PortEntity.AllowAddressPair> newAllowedAddressPairs = newPortEntity.getAllowedAddressPairs();
+        List<PortEntity.AllowAddressPair> oldAllowedAddressPairs = oldPortEntity.getAllowedAddressPairs();
+
+        if (newAllowedAddressPairs != null && !newAllowedAddressPairs.equals(oldAllowedAddressPairs)) {
+            oldPortEntity.setAllowedAddressPairs(newAllowedAddressPairs);
+        }
+    }
+
+    private void updatePortSecurityEnabled(PortEntity newPortEntity, PortEntity oldPortEntity) {
+        boolean newPortSecurityEnabled = newPortEntity.isPortSecurityEnabled();
+        boolean oldPortSecurityEnabled = oldPortEntity.isPortSecurityEnabled();
+
+        if (newPortSecurityEnabled != oldPortSecurityEnabled) {
+            oldPortEntity.setPortSecurityEnabled(newPortSecurityEnabled);
+        }
+    }
+
+    private void updateQosPolicyId(PortEntity newPortEntity, PortEntity oldPortEntity) {
+        String newQosPolicyId = newPortEntity.getQosPolicyId();
+        String oldQosPolicyId = oldPortEntity.getQosPolicyId();
+
+        if (newQosPolicyId != null && !newQosPolicyId.equals(oldQosPolicyId)) {
+            oldPortEntity.setQosPolicyId(newQosPolicyId);
+        }
+    }
+
+
+    private void updateMacLearningEnabled(PortEntity newPortEntity, PortEntity oldPortEntity) {
+        boolean newMacLearningEnabled = newPortEntity.isMacLearningEnabled();
+        boolean oldMacLearningEnabled = oldPortEntity.isMacLearningEnabled();
+
+        if (newMacLearningEnabled != oldMacLearningEnabled) {
+            oldPortEntity.setMacLearningEnabled(newMacLearningEnabled);
+        }
+    }
+
+    private InternalPortEntity buildInternalPortEntity(PortEntity portEntity) {
+        InternalPortEntity internalPortEntity =
+                new InternalPortEntity(portEntity, null, null, null);
+        NetworkConfig.ExtendPortEntity extendPortEntity =
+                new NetworkConfig.ExtendPortEntity(internalPortEntity, portEntity.getBindingHostId());
+
+        return extendPortEntity;
+    }
+
+    private void buildInternalPortEntities(PortContext context, List<PortEntity> portEntities) {
+        List<InternalPortEntity> internalPortEntities = new ArrayList<>();
+
+        for (PortEntity portEntity: portEntities) {
+            if (portEntity.getBindingHostId() != null) {
+                internalPortEntities.add(buildInternalPortEntity(portEntity));
+            }
+        }
+
+        context.getNetworkConfig().setPortEntities(internalPortEntities);
+    }
+
+    @Override
+    void createProcess(PortContext context) {
```

Comment:
This createProcess only applies to port with binding host id. How about those without binding host id?